src/Microsoft.OpenApi/Validations/OpenApiElementValidator.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. 
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.OpenApi.Interfaces;
+using Microsoft.OpenApi.Validations.Rules;
+
+namespace Microsoft.OpenApi.Validations
+{
+    /// <summary>
+    /// The public APIs to validate the Open API element.
+    /// </summary>
+    public static class OpenApiElementValidator
+    {
+        /// <summary>
+        /// Validate the Open API element.
+        /// </summary>
+        /// <typeparam name="T">The Open API element type.</typeparam>
+        /// <param name="element">The Open API element.</param>
+        /// <returns>True means no errors, otherwise with errors.</returns>
+        public static bool Validate<T>(this T element) where T : IOpenApiElement
+        {
+            IEnumerable<ValidationError> errors;
+            return element.Validate(out errors);
+        }
+
+        /// <summary>
+        /// Validate the Open API element.
+        /// </summary>
+        /// <typeparam name="T">The Open API element type.</typeparam>
+        /// <param name="element">The Open API element.</param>
+        /// <param name="errors">The output errors.</param>
+        /// <returns>True means no errors, otherwise with errors.</returns>
+        public static bool Validate<T>(this T element, out IEnumerable<ValidationError> errors)
+            where T : IOpenApiElement
+        {
+            ValidationRuleSet ruleSet = ValidationRuleSet.DefaultRuleSet;
+            return element.Validate(ruleSet, out errors);
+        }
+
+        /// <summary>
+        ///  Validate the Open API element.
+        /// </summary>
```

Comment:
Ok, so here's the deal!  If I have to put up with these horrible comment blocks, I would like to see comments that actually justify their existence.  So let's make the comment explain why I would pick one overload over the other so that I don't have to guess the reason based on what parameters are there.

Reply:
Great point. I will modify it.

---
In reply to: [157324617](https://github.com/Microsoft/OpenAPI.NET/pull/152#discussion_r157324617) [](ancestors = 157324617)

Reply:
modified.

---
In reply to: [157324939](https://github.com/Microsoft/OpenAPI.NET/pull/152#discussion_r157324939) [](ancestors = 157324939,157324617)